win32/Lib/win32timezone.py

```diff
@@ -585,30 +613,32 @@ def _LoadDynamicInfoFromKey(self, key):
             key_match_comparator=operator.ge,
         )
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         result = f"{self.__class__.__name__}({repr(self.timeZoneName)}"
         if self.fixedStandardTime:
             result += ", True"
         result += ")"
         return result
 
-    def __str__(self):
+    def __str__(self) -> str:
         return self.displayName
 
-    def tzname(self, dt):
-        winInfo = self.getWinInfo(dt)
+    def tzname(self, dt: datetime.datetime | None) -> str:
+        # FIXME/TODO: Real issue !
+        # Are we meant to return the last item if passing a datetime to getWinInfo?
+        winInfo = self.getWinInfo(dt)  # type: ignore[arg-type]
```

Comment:
This is an issue that still needs to be solved. I'm not certain what exactly was intended here. CCing @jaraco as the author

Reply:
> Are we meant to return the last item if passing a datetime to getWinInfo?

I don't fully understand the question:

I see that `.tzname` isn't tested anywhere. I'd suggest we start with that.

This method is implemented as required by the [interface](https://docs.python.org/3/library/datetime.html#datetime.tzinfo.tzname). It is indeed supposed to accept a `datetime.datetime`, so it's probably an error that it's passing it to `getWinInfo` that requires an integer year. Probably it should just be:

```suggestion
        winInfo = self.getWinInfo(dt.year)
```

Wait! I started testing on my Windows machine using the latest stable pywin32 (306), and it seems to resolve `dt.year` (line 637 references `getWinInfo(dt.year)`.

In b66d9797, I added a test and fixed the implementation so the tests pass. Feel free to cherry-pick that in here, or if you or Mark would rather it be a separate PR, I can do that.

Reply:
I think a separate PR would be lovely.
I'll probably split my `__init_from_other` fix into a separate (simple to review) PR as well.
Same for the `RangeMap` changes to be pulled back from https://github.com/jaraco/jaraco.collections/pull/16 once that's completed.

Reply:
Resolved in #2337